backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/codec/impl/LuceneGeoPointFieldCodec.java

```diff
@@ -54,9 +54,13 @@ public void encode(LuceneDocumentBuilder documentBuilder, String absoluteFieldPa
 			documentBuilder.addField( new StoredField( getLongitudeAbsoluteFieldPath( absoluteFieldPath ), value.getLongitude() ) );
 		}
 
-		if ( searchable || sortable ) {
-			// doc values fields are required for predicates, distance projections and distance sorts
+		if ( sortable || searchable ) {
+			// Generally the LatLonDocValuesField field is required only for sorting,
+			// but we rely on this field to support exists predicate too.
```

Comment:
To address the `exists` predicate, see how we do it for numeric fields.

In particular, when encoding:

```
		if ( sortable ) {
			documentBuilder.addField( domain.createDocValuesField( absoluteFieldPath, encodedValue ) );
		}
		else {
			// For createExistsQuery()
			documentBuilder.addFieldName( absoluteFieldPath );
		}
```

and when creating the `exists` query:

```
		if ( sortable ) {
			return new DocValuesFieldExistsQuery( absoluteFieldPath );
		}
		else {
			return new TermQuery( new Term( LuceneFields.fieldNamesFieldName(), absoluteFieldPath ) );
		}
```

Reply:
I think that it's a nice solution. I applied it.
I had only to add an `| projectable` since it seems that a `DocValuesField` is required by the our `DistanceCollector` to project a distance form the very point.